tokio/src/sync/mpsc/bounded.rs

```diff
@@ -150,6 +152,43 @@ impl<T> Receiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Blocking receive to call outside of asynchronous contexts
+    ///
+    /// ```
+    /// use std::thread;
+    /// use tokio::runtime::Runtime;
+    /// use tokio::sync::mpsc;
+    ///
+    /// fn main() {
+    ///     let (mut tx, mut rx) = mpsc::channel::<u8>(10);
+    ///     
+    ///     let sync_code = thread::spawn(move || {
+    ///         assert_eq!(Ok(Some(10)), rx.blocking_recv());
+    ///     });
+    ///
+    ///     Runtime::new()
+    ///         .unwrap()
+    ///         .block_on(async move {
+    ///             let _ = tx.send(10).await;
+    ///         });
+    ///     sync_code.join().unwrap()
+    /// }
+    /// ```
+    #[cfg(feature = "blocking")]
+    pub fn blocking_recv(&mut self) -> Result<Option<T>, ()> {
+        let _enter_handle = crate::runtime::enter::enter(false);
+        let mut park = CachedParkThread::new();
```

Comment:
Instead of implementing the blocking logic, you should be able to use `enter_handle.block_on(&self.recv())`.

You will need to add `sync` to [this](https://github.com/tokio-rs/tokio/blob/646fbae76535e397ef79dbcaacb945d4c829f666/tokio/src/macros/cfg.rs#L59-L67)

Reply:
I guess I should also use block_on for the `blocking_send` implementation as well?

Reply:
@carllerche using block_on broke the features ci run, so I added a `cfg_block_on` around both methods. If it's still not working or there's any other comments I'll be around tomorrow morning to resolve them :smile: